engine/src/main/java/org/hibernate/search/cfg/spi/SearchConfigurationBase.java

```diff
@@ -60,4 +72,46 @@ public boolean isIdProvidedImplicit() {
 	public boolean isMultitenancyEnabled() {
 		return false;
 	}
+
+	@Override
+	public ErrorHandler getErrorHandler() {
+		if ( errorHandler == null ) {
+			errorHandler = createErrorHandler();
+		}
+		return errorHandler;
+	}
+
+	private ErrorHandler createErrorHandler() {
```

Comment:
I'd prefer to keep the logic in `SearchConfiguration` implementations to the bare minimal, so that it's easy to create a different implementation.

Could you undo moving these methods, and simply have the `getErrorHandler()` of this implementation return null?